### ISSUE TITLE
Doesn't quote colons

### ISSUE THREAD

**Issue by rix0rrr** [eemeli/yaml#43]
Not sure about the exact spec (if more special characters need to be quoted and maybe in other combinations), but at least this library doesn't quote single colons, which leads it to generate invalid YAML:

```
$ npm init
...
$ npm install yaml
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN pkg@1.0.0 No description
npm WARN pkg@1.0.0 No repository field.

+ yaml@1.0.0
added 1 package in 0.825s

$ node -p "require('yaml').stringify({ key: ':' })"
key: :

# PyYaml doesn't like it:

$ python3 -c 'import yaml; from io import StringIO; print(yaml.load(StringIO("key: :")))'
yaml.scanner.ScannerError: mapping values are not allowed here
  in "<file>", line 1, column 6
```

**Comment by eemeli** [comment]
That is indeed a bug. To verify, even we complain about it:
```js
YAML.parse(YAML.stringify({ key: ':' }))
// YAMLSemanticError: Nested Mappings are not allowed in compact mappings
```

**Comment by rix0rrr** [comment]
Thanks for the speedy fix!

**Comment by eemeli** [comment]
Fix released in version 1.0.1.
